### DISCUSSION TITLE
Variable Circle Size for Osu Standard Gameplay

### DISCUSSION THREAD

**Discussion by Fizzxx** [ppy/osu#13028] [Ideas]
**Describe the new feature:** 
With the new release, I would like to see a mod that could change the circle size as you play. The size dependency could be on health, time, etc. This could enable players to practice improving their accuracy on songs without being restricted to the size change offered by the Hard Rock mod, or even just be a nice addition to "Fun" category of mods available to the game. With the right curve, I believe the mod could be as visually appealing as it may be helpful. 

**Proposal designs of the feature:** 
I have done some slight modifications to the existing mod "Grow" to try and make this mod a reality. It simply introduces a scale factor that is calculated based on the current health value, and then applies it to the hit objects as the health value updates. The size of the hit objects is allowed to change between 2x (lower hp), and 0.5x (higher hp), it's default size for any particular beatmap. The curve for the scale factor is currently set to 2^-(2*hp - 1), allowing for a (mostly) steady/smooth transition between the maximum and minimum circle size.

I believe in the future (after some successful testing), a more appropriate curve could be implemented, or perhaps a means of standardizing the circle size (under the effects of the mod, of course) across all beatmaps. Since 2x the circle size at circle size 2, or even 0.5x the circle size at circle size 6+ may be a bit unreasonable for gameplay. 

**Roadblocks I've encountered:**
As stated above, I have already begun trying to create a working version of the mod. What I have so far can be found here: https://github.com/Fizzxx/osuLazer . The main issue I'm running into is that with each instance of the mod, the scale factor (radiusScaleFactor) is being reset to the value it is declared at. This causes the hit objects to remain fixed at that scale factor until the moment at the end of the hitobject's lifetime, at which point it will instantaneously shrink/grow to the desired size (according to the current hp) during its hit/miss animation. I am by no means what I would call "proficient" in coding/programming, so any questions and/or feedback are more than welcome, and thank you in advance!

**Comment by LeNitrous** [comment]
per-note difficulty settings are planned if my memory serves right

**Comment by iSimpFor** [comment]
i think you can also apply this to catch the beat(gamemode)
